.github/workflows/tf-test-compatability.yml

```diff
@@ -0,0 +1,75 @@
+on: workflow_call
+
+jobs:
+  get-versions:
+    runs-on: ubuntu-latest
+  outputs:
+    tfl3: ${ steps.tfl-3.outputs.tfl3 }
+  steps:
+    - name: Install tfenv
+      shell: bash
+      run: |
+        git clone --depth=1 https://github.com/tfutils/tfenv.git tfenv
+
+    - name: TerraformLatest-3
+      id: tfl-3
+      shell: bash
+      run: |
+        echo "tfl3=$(tfenv/bin/tfenv list-remote | grep '1\.[0-9]\.0$' | sed -n '4p')" >> $GITHUB_OUTPUT
+
+  run-tests:
+    needs: get-versions
+    strategy:
+      matrix:
+        version: [${{ needs.get-versions.outputs.tfl3 }}]    
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::249446771485:role/tf-role-arc-generic
```

Comment:
I had this as a secret before but I need to set that secret on every single one of the terraform repos (or use an org level secret, which I'm not sure Salim will want to do)

Since this already in plaintext in the go-sdk repo and Salim said it was ok when I raised it with him I chose to save myself some work adding this as a secret and just put it in plaintext as it already is in the go-sdk repo.